### ISSUE TITLE
Bookie can't start using 4.7.2 docker image

### ISSUE THREAD

**Issue by sijie** [apache/bookkeeper#1660]
**BUG REPORT**

1. Please describe the issue you observed:

- What did you do?

upgrade to use bookkeeper 4.7.2 docker image.

- What did you expect to see?

bookie can start on a new environment.

- What did you see instead?

bookie failed to start.

```
2018-09-06 06:53:51,124 - ERROR - [main:Main@221] - Failed to build bookie server
org.apache.bookkeeper.bookie.BookieException$MetadataStoreException: Failed to get cluster instance id
	at org.apache.bookkeeper.discover.ZKRegistrationManager.getClusterInstanceId(ZKRegistrationManager.java:387)
	at org.apache.bookkeeper.bookie.Bookie.checkEnvironmentWithStorageExpansion(Bookie.java:412)
	at org.apache.bookkeeper.bookie.Bookie.checkEnvironment(Bookie.java:256)
	at org.apache.bookkeeper.bookie.Bookie.<init>(Bookie.java:640)
	at org.apache.bookkeeper.proto.BookieServer.newBookie(BookieServer.java:131)
	at org.apache.bookkeeper.proto.BookieServer.<init>(BookieServer.java:100)
	at org.apache.bookkeeper.server.service.BookieService.<init>(BookieService.java:43)
	at org.apache.bookkeeper.server.Main.buildBookieServer(Main.java:299)
	at org.apache.bookkeeper.server.Main.doMain(Main.java:219)
	at org.apache.bookkeeper.server.Main.main(Main.java:201)
Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for BookKeeper metadata
	at org.apache.bookkeeper.discover.ZKRegistrationManager.getClusterInstanceId(ZKRegistrationManager.java:377)
	... 9 more
```

**Comment by sijie** [comment]
The problem is related to #1601 . in 3.4.13, zookeeper cli always returns '0' as return code, which cause the docker image skip initializing cluster.  zookeeper commit: fade262a8a4fa744bd768dc7569d8f4c9d84f275 and issue [ZOOKEEPER-1898](https://issues.apache.org/jira/browse/ZOOKEEPER-1898). 

the problem will only be caught when a cluster is never initialized by the bookkeeper shell script before. will see how to fix it (or work around in the docker image) and add a test case to cover this case.

See discussion at https://github.com/apache/bookkeeper/pull/1644#issuecomment-418991073

**Comment by sijie** [comment]
This is fixed by #1666
